### PR TITLE
Julia

### PR DESCRIPTION
**Pull request by JKRT**
@sjoelund 
Commit spam for now I hope that it is ok. I have started work on a testsuite it depends on MetaModelica.jl 

My idea was to divide it up into syntax tests and semantic tests with the same tests I wrote for my LLVM stuff 

(I am far from an expert in Julia so if you have some comments on it please leave them so I learn)

### CODE REVIEW

**Comment by sjoelund** [comment]
Seems OK so far from skimming the changes. New template AbsynToJulia and some changes to utility functions to make things work I suppose.

File: OMCompiler/toJulia/lexer.jl [hunk 1]
```diff
@@ -15,10 +15,11 @@ keywordWithAction = [
 string   = re"\"([^\"\\x5c]|(\\x5c['\"?\\x5cabfnrtv]))*\""
 ident    = re"[_A-Za-z][_A-Za-z0-9]*|'([^'\\x5c]|(\\x5c.))+'"
 int      = re"[0-9]+"
+bool     = re "(tru|fals)e"
```
Comment (JKRT) [inline OMCompiler/toJulia/lexer.jl:18]: @sjoelund Is this correct? I am not sure.. my regexp skills are not exactly 100%

File: OMCompiler/toJulia/tokens.jl [hunk 2]
```diff
@@ -1,15 +1,19 @@
...
 struct IntToken <: Token
-  value::Int
+  value::Signed
 end
 struct RealToken <: Token
   original::String
-  value::Real
+  value::ModelicaReal
+end
+struct BoolToken <: Token
```
Comment (JKRT) [inline OMCompiler/toJulia/tokens.jl:15]: @sjoelund  When I added this I thought it was necessary, but everything is  still more or less WIP

**Review by JKRT — COMMENTED** [review]
Some minor comments

**Comment by JKRT** [comment]
> 
> 
> Seems OK so far from skimming the changes. New template AbsynToJulia and some changes to utility functions to make things work I suppose.

More or less, what does not work (It is really quite many things that do not work) in the AbsynToJulia.tpl are Uniontypes, Absyn also surprises me a little bit by giving me Modelica records when I ought to receive MM_Records. Is this imagination or is it an issue with Absyn?

File: OMCompiler/toJulia/MMToJuliaTestSuite/syntaxCheck.jl [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+end
+
+function executeTestSteps(homeDirectory, sourceDirectory, outputDirectory, omc)
+  outDir = createDirectoryReportErrorOnFailure(abspath(outputDirectory))
+  translateFilesIfOutputIsEmpty(sourceDirectory, outDir, omc, homeDirectory)
+  checkSyntax(outDir, sourceDirectory)
+  cd(homeDirectory)
+end
+
+function syntaxCheck(omc)
```
Comment (JKRT) [inline OMCompiler/toJulia/MMToJuliaTestSuite/syntaxCheck.jl:55]: This test currently does two things. It both tests the syntax and if AbsynToJulia.tpl translates without throwing errors. Currently, my algorithms test is failing. I plan to add a test for translating Graphviz at some point when ./Algorithm is translated correctly. 

When the syntax tests passes I will add semantic tests that tests these different functions

File: OMCompiler/toJulia/lexer.jl (hunk 1 above)
Comment (sjoelund) [reply OMCompiler/toJulia/lexer.jl:18]: Should be correct

File: OMCompiler/toJulia/tokens.jl (hunk 2 above)
Comment (sjoelund) [reply OMCompiler/toJulia/tokens.jl:15]: Probably not necessary as it's just tokens

**Comment by sjoelund** [comment]
Yes, Absyn does lack some of the uniontype things. It's stored just like it is in the source code (regular record and only inside the uniontype). Can probably be handled in some nice way for uniontypes (asserting that uniontype containing records are handled in a different way). The main difficulty might be for some of the packages that are actually uniontypes like NFExpression.mo.

**Pull request closed without merging**